src/objectcontroller.cpp

```diff
@@ -1478,6 +1484,303 @@ void DateObject::getDate(int *day, int *month, int *year)
+        return 1;
+    if (hour_m < hour)
+        return -1;
+
+    if (min_m > min)
+        return 1;
+    if (min_m < min)
+        return -1;
+
+    if (sec_m > sec)
+        return 1;
+    if (sec_m < sec)
+        return -1;
+
+    return 0;
+}
+
+bool DateTimeObjectValue::set(ObjectValue* value)
+{
+    int day, month, year, wday, hour, min, sec;
+    assert(value);
+    DateTimeObjectValue* val = dynamic_cast<DateTimeObjectValue*>(value);
+    if (val == 0)
+        logger_m.errorStream() << "DateTimeObjectValue: ERROR, setValue() received invalid class object (typeid=" << typeid(*value).name() << ")" << endlog;
+    else
+    {
+        val->getDateTimeValue(&day, &month, &year, &wday, &hour, &min, &sec);
+        if ( day_m != day || month_m != month || year_m != year ||
+	     wday_m != wday || hour_m != hour || min_m != min || sec_m != sec )
+        {
+            day_m = day;
+            month_m = month;
+            year_m = year;
+            wday_m = wday;
+            hour_m = hour;
+            min_m = min;
+            sec_m = sec;
+            return true;
+        }
+    }
+    return false;
+}
+
+bool DateTimeObjectValue::set(double value)
+{
+    int date, day, month, year, wday, hour, min, sec;
+    if (value < 0)
+    {
+        day = -1;
+        month = -1;
+        year = -1;
+        wday = -1;
+        hour = -1;
+        min = -1;
+        sec = -1;
+    }
+    else
+    {
+	date = value / 86400;
+	value -= date * 86400;
+
+        year = date / 400;
+        date -= date * 400;
+        month = date / 31;
+        day = date - month * 31;
+
+	wday = 0;
+
+	hour = value / 3600;
+	value -= hour * 3600;
+	min = value / 60;
+	sec = value - min * 60;
+    }
+    if ( day_m != day || month_m != month || year_m != year|| wday_m != wday ||  hour_m != hour || min_m != min || sec_m != sec )
+    {
+        day_m = day;
+        month_m = month;
+        year_m = year;
+        wday_m = wday;
+        hour_m = hour;
+        min_m = min;
+        sec_m = sec;
+        return true;
+    }
+    return false;
+}
+
+void DateTimeObjectValue::getDateTimeValue(int *day, int *month, int *year, int* wday, int *hour, int *min, int *sec)
+{
+    if (day_m == -1)
+    {
+        time_t t = time(0);
+        struct tm * timeinfo = localtime(&t);
+        *day = timeinfo->tm_mday;
+        *month = timeinfo->tm_mon+1;
+        *year = timeinfo->tm_year;
+        *wday = timeinfo->tm_wday;
+	if (*wday == 0)
+          *wday = 7;
+        *hour = timeinfo->tm_hour;
+        *min = timeinfo->tm_min;
+        *sec = timeinfo->tm_sec;
+    }
+    else
+    {
+        *day = day_m;
+        *month = month_m;
+        *year = year_m;
+        *wday = wday_m;
+        *hour = hour_m;
+        *min = min_m;
+        *sec = sec_m;
+    }
+}
+
+Logger& DateTimeObject::logger_m(Logger::getInstance("DateTimeObject"));
+
+DateTimeObject::DateTimeObject() : DateTimeObjectValue(0, 0, 0, 0, 0, 0, 0)
+{}
+
+DateTimeObject::~DateTimeObject()
+{}
```

Comment:
Can you inline them in the header, since they are empty? That would be a few less lines in this huge cpp file :+1: